File: Atcoder/ABC116/C/c.cpp
#include <iostream>
#include <vector>
#include <map>
#include <cstdio>
#include <string>
#include <algorithm>
using namespace std;

int main(void){
    int N;
    cin >> N;
    vector<int> h = vector<int>(N, 0);
    for(int cnt = 0; cnt < N; cnt++){
        cin >> h[cnt];
    }
    int result = 0;
    int count = 1;
    while(count < 101){
        bool once = false;
        bool continuous = false;
        for(int cnt = 0; cnt < N; cnt++){
            if(h[cnt] >= count){
                continuous = true;
                once = true;
            }else{
                if(continuous){
                    result++;
                }
                continuous = false;
            }
        }
        if(!once){
            break;
        }else if(continuous){
            result++;
        }
        count++;
    }
    cout << result << endl;

    return 0;
}

File: AtCoder/ABC116/C/c.rb
# encoding : utf-8

N = gets.to_i
h = gets.split.map(&:to_i)
result, count = 0, 1
until count > 100
    once = false
    continuous =false
    N.times do |cnt|
        if h[cnt] >= count
            continuous = true
            once = true
        else
            if continuous
                result += 1
            end
            continuous = false
        end
    end
    if !once
        break
    elsif continuous
        result += 1
    end
    count += 1
end
puts result
